packages/openapi3/test/union-schema.test.ts

```diff
@@ -682,6 +681,82 @@ worksFor(["3.0.0"], ({ diagnoseOpenApiFor, oapiForModel, openApiFor }) => {
         message: "Cannot have a union containing only null types.",
       });
     });
+
+    it("should keep original type with null", async () => {
+      const openApi = await openApiFor(`
+        scalar Str extends string;
+        model Mol {x: string}
+        /** this is my Num */
+        scalar Num extends int32;
+        scalar More extends Num;
+        scalar Int16 extends int16;
+        scalar NoExtends;
+        scalar NoRoot extends NoExtends;
+        
+        model Test {
+          @minValue(1)
+          a: NoRoot | NoExtends | Str | Int16 | Num | Mol | More | null | int32 | string;
+        }
+        
+        @minValue(1)
+        union u {
+        NoRoot, NoExtends, Str , Int16 , Num , Mol , More , null , int32 , string
+        }
+      `);
+
+      expect(openApi.components.schemas.Num).toEqual({
+        type: "integer",
+        format: "int32",
+        description: "this is my Num",
+      })
+
+      expect(openApi.components.schemas.More).toEqual({
+        type: "integer",
+        format: "int32",
+        description: "this is my Num",
+      })
+
+      expect(openApi.components.schemas.Test).toEqual({
```

Comment:
please split those test into individual ones that describe what they are testing, this is here a massive smoke test which we don't know if we it was by design or just accepted.

Reply:
and try to make the test smaller than including a dozen types, only the minimum needed to describve the behavior